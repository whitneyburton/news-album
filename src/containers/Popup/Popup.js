import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

export class Popup extends Component {

  findPopup = () => {
    const { id } = this.props.match.params;
    const { articles } = this.props;
    const popup = articles.find(article => article.id === id);
    return popup;
  }
  
  render() {
    const article = this.findPopup();
    return (
      <div className='Popup'>
        {article.title}
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  articles: state.articles
})

export default connect(mapStateToProps)(Popup);

Popup.propTypes = {
  id: PropTypes.string,
  articles: PropTypes.array
}