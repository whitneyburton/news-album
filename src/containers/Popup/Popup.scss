.Popup {
  display: flex;
  justify-content: space-between;
  background-color: #1f1f1f;
  width: 95%;
  height: 70%;
  position: fixed;
  z-index: 1;
  margin: auto;
  top: 21%;
  color: #fff;
  left: 2.4%;

  .Popup--info {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    padding: 0px 40px;
    letter-spacing: 1px;
    font-size: 20px;
    font-weight: 300;
  }

  .Popup--facts {
    padding: 15px;
  }

  .Popup--author-pub {
    padding: 4% 0%;
  }

  .Popup--title {
    font-size: 2em;
    padding-top: 2%;
  }

  .Popup--image {
    width: 50%;
  }

  .Popup--star {
    position: absolute;
    top: 2%;
    background-color: #1f1f1f;
    padding: 10px 12px;
    border-radius: 50%;
    left: 94%;
    cursor: pointer;
    transition: all 500ms ease;

    &:hover {
      background-color: #424242;
    }
  }

  .Popup--close {
    position: absolute;
    background-color: #1f1f1f;
    top: 2%;
    padding: 10px 12px;
    border-radius: 50%;
    left: 88%;
    cursor: pointer;
    color: #000;
    transition: all 500ms ease;

    &:hover {
      background-color: #424242;
    }
  }
}

.Popup--buttons {
  margin-left: auto;
}

button {
  background-color: #727272;
  color: #fff;
  border: none;
  width: 130px;
  height: 42px;
  border-radius: 20px 20px 0px 0px;
  letter-spacing: 4px;
  font-weight: 500;
  font-size: 18px;
  margin-right: 20px;
  cursor: pointer;
  transition: all 500ms ease;

  &:hover {
    background-color: #424242;
  }

  &:focus {
    outline: none;
  }
}

@media screen and (max-width: 1100px) {
  .Popup {
    flex-direction: column;
    display: flex;
    top: 10%;
    left: 1.4%;
    width: 97%;
    height: 80%;

    .Popup--info {
      overflow: scroll;
      font-size: 16px;
      padding-top: 0;
    }

    .Popup--author-pub {
      padding: 0%;
    }

    .Popup--facts {
      padding: 0px;
    }

    .Popup--title {
      font-size: 1.4em;
      padding-top: 2%;
    }

    .Popup--delete-fav {
      display: flex;
      justify-content: space-evenly;
      padding-top: 10px;

      .Popup--star {
        position: unset;
        background-color: #727272;
      }

      .Popup--close {
        position: unset;
        background-color: #727272;
      }
    }

    .Popup--buttons {
      text-align: center;
      margin: 0;
    }

    .Popup--image {
      margin: auto;
      height: 54%;
    }

    button {
      border-radius: 20px;
      font-size: 13px;
      width: 35%;
      height: 30px;
      margin-top: 10px;
      font-weight: 400;
    }
  }
}

@media screen and (max-width: 600px) {
  .Popup {
    .Popup--image {
      width: 75%;
    }
  }
}

@media screen and (max-width: 480px) {
  .Popup {
    .Popup--image {
      height: 45%;
    }
  }
}
