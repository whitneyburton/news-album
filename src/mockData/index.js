export const mockNatGeoArticle = {
  author: 'Douglas Main',
  title: 'Why insect populations are plummeting',
  description: 'Sad insects.',
  url:
    'https://www.nationalgeographic.com/animals/2019/02/why-insect-populations-are-plummeting-and-why-it-matters.html',
  urlToImage:
    'https://www.nationalgeographic.com/content/dam/animals/2019/02/insects-in-decline/01-insects-in-decline-nationalgeographic_1353541.jpg',
  publishedAt: '2019-02-19T02:07:15.6368398Z',
  id: 'b',
  isFavorite: false
};

export const mockCryptoCoinsArticle = {
  author: 'Greg Thompson',
  title: 'Ethereum Slayers',
  description: 'They are everywhere.',
  url: 'https://www.ccn.com/ethereum-killers-crypto-blockchain',
  urlToImage:
    'https://www.ccn.com/wp-content/uploads/2019/02/ethereum-crypto-dark-cement-shutterstock.jpg',
  publishedAt: '2019-02-23T14:28:21Z',
  id: 'f',
  isFavorite: false
};

export const mockNewScientistArticle = {
  author: 'Leah Crane',
  title: 'Four stars full of burnt nuclear ash',
  description: 'Scary stars!',
  url:
    'https://www.newscientist.com/article/2194683-four-stars-full-of-burnt-nuclear-ash-are-speeding-through-the-galaxy/',
  urlToImage:
    'https://images.newscientist.com/wp-content/uploads/2019/02/22150921/vennes1hr.jpg',
  publishedAt: '2019-02-22T00:00:00Z',
  id: 'd',
  isFavorite: false
};

export const mockNatGeoArticles = [
  {
    author: 'Erica Johnson',
    title: '20 U.S. state parks',
    description: 'Pretty parks!',
    url:
      'https://www.nationalgeographic.com/travel/destinations/north-america/united-states/state-parks-worth-the-trip.html',
    urlToImage:
      'https://www.nationalgeographic.com/content/dam/travel/2019-digital/state-parks-to-know/02-state-parks-waianapanapa-park-hawaii.jpg',
    publishedAt: '2019-02-19T02:07:16.9805881Z',
    id: 'a',
    isFavorite: false
  },
  {
    author: 'Douglas Main',
    title: 'Why insect populations are plummeting',
    description: 'Sad insects.',
    url:
      'https://www.nationalgeographic.com/animals/2019/02/why-insect-populations-are-plummeting-and-why-it-matters.html',
    urlToImage:
      'https://www.nationalgeographic.com/content/dam/animals/2019/02/insects-in-decline/01-insects-in-decline-nationalgeographic_1353541.jpg',
    publishedAt: '2019-02-19T02:07:15.6368398Z',
    id: 'b',
    isFavorite: false
  }
];

export const mockNatGeoArticlesWithFav = [
  {
    author: 'Erica Johnson',
    title: '20 U.S. state parks',
    description: 'Pretty parks!',
    url:
      'https://www.nationalgeographic.com/travel/destinations/north-america/united-states/state-parks-worth-the-trip.html',
    urlToImage:
      'https://www.nationalgeographic.com/content/dam/travel/2019-digital/state-parks-to-know/02-state-parks-waianapanapa-park-hawaii.jpg',
    publishedAt: '2019-02-19T02:07:16.9805881Z',
    id: 'a',
    isFavorite: false
  },
  {
    author: 'Douglas Main',
    title: 'Why insect populations are plummeting',
    description: 'Sad insects.',
    url:
      'https://www.nationalgeographic.com/animals/2019/02/why-insect-populations-are-plummeting-and-why-it-matters.html',
    urlToImage:
      'https://www.nationalgeographic.com/content/dam/animals/2019/02/insects-in-decline/01-insects-in-decline-nationalgeographic_1353541.jpg',
    publishedAt: '2019-02-19T02:07:15.6368398Z',
    id: 'b',
    isFavorite: true
  }
];

export const mockNewScienceArticles = [
  {
    author: 'Chelsea Whyte',
    title: 'Wolves are spreading in Germany',
    description: 'Wolves everywhere!',
    url:
      'https://www.nationalgeographic.https://www.newscientist.com/article/2194696-wolves-are-spreading-in-germany-by-moving-into-military-land//animals/2019/02/why-insect-populations-are-plummeting-and-why-it-matters.html',
    urlToImage:
      'https://images.newscientist.com/wp-content/uploads/2019/02/22145340/rm7694.jpg',
    publishedAt: '2019-02-23T00:00:00Z',
    id: 'c',
    isFavorite: false
  },
  {
    author: 'Leah Crane',
    title: 'Four stars full of burnt nuclear ash',
    description: 'Scary stars!',
    url:
      'https://www.newscientist.com/article/2194683-four-stars-full-of-burnt-nuclear-ash-are-speeding-through-the-galaxy/',
    urlToImage:
      'https://images.newscientist.com/wp-content/uploads/2019/02/22150921/vennes1hr.jpg',
    publishedAt: '2019-02-22T00:00:00Z',
    id: 'd',
    isFavorite: false
  }
];

export const mockNewScienceArticlesWithFav = [
  {
    author: 'Chelsea Whyte',
    title: 'Wolves are spreading in Germany',
    description: 'Wolves everywhere!',
    url:
      'https://www.nationalgeographic.https://www.newscientist.com/article/2194696-wolves-are-spreading-in-germany-by-moving-into-military-land//animals/2019/02/why-insect-populations-are-plummeting-and-why-it-matters.html',
    urlToImage:
      'https://images.newscientist.com/wp-content/uploads/2019/02/22145340/rm7694.jpg',
    publishedAt: '2019-02-23T00:00:00Z',
    id: 'c',
    isFavorite: false
  },
  {
    author: 'Leah Crane',
    title: 'Four stars full of burnt nuclear ash',
    description: 'Scary stars!',
    url:
      'https://www.newscientist.com/article/2194683-four-stars-full-of-burnt-nuclear-ash-are-speeding-through-the-galaxy/',
    urlToImage:
      'https://images.newscientist.com/wp-content/uploads/2019/02/22150921/vennes1hr.jpg',
    publishedAt: '2019-02-22T00:00:00Z',
    id: 'd',
    isFavorite: true
  }
];

export const mockCryptoCoinsArticles = [
  {
    author: 'Joseph Young',
    title: 'Bitcoins spike',
    description: 'So much money!',
    url:
      'https://www.ccn.com/bitcoin-price-4100-2-hours-analyst-crypto-bullish',
    urlToImage:
      'https://www.ccn.com/wp-content/uploads/2019/02/bitcoin-price-green-chart-bull-shutterstock.jpg',
    publishedAt: '2019-02-23T14:45:23Z',
    id: 'e',
    isFavorite: false
  },
  {
    author: 'Greg Thompson',
    title: 'Ethereum Slayers',
    description: 'They are everywhere.',
    url: 'https://www.ccn.com/ethereum-killers-crypto-blockchain',
    urlToImage:
      'https://www.ccn.com/wp-content/uploads/2019/02/ethereum-crypto-dark-cement-shutterstock.jpg',
    publishedAt: '2019-02-23T14:28:21Z',
    id: 'f',
    isFavorite: false
  }
];

export const mockCryptoCoinsArticlesWithFav = [
  {
    author: 'Joseph Young',
    title: 'Bitcoins spike',
    description: 'So much money!',
    url:
      'https://www.ccn.com/bitcoin-price-4100-2-hours-analyst-crypto-bullish',
    urlToImage:
      'https://www.ccn.com/wp-content/uploads/2019/02/bitcoin-price-green-chart-bull-shutterstock.jpg',
    publishedAt: '2019-02-23T14:45:23Z',
    id: 'e',
    isFavorite: false
  },
  {
    author: 'Greg Thompson',
    title: 'Ethereum Slayers',
    description: 'They are everywhere.',
    url: 'https://www.ccn.com/ethereum-killers-crypto-blockchain',
    urlToImage:
      'https://www.ccn.com/wp-content/uploads/2019/02/ethereum-crypto-dark-cement-shutterstock.jpg',
    publishedAt: '2019-02-23T14:28:21Z',
    id: 'f',
    isFavorite: true
  }
];

export const mockNatGeoArticlesObj = {
  articles: [
    {
      author: 'Erica Johnson',
      title: '20 U.S. state parks',
      description: 'Pretty parks!',
      url:
        'https://www.nationalgeographic.com/travel/destinations/north-america/united-states/state-parks-worth-the-trip.html',
      urlToImage:
        'https://www.nationalgeographic.com/content/dam/travel/2019-digital/state-parks-to-know/02-state-parks-waianapanapa-park-hawaii.jpg',
      publishedAt: '2019-02-19T02:07:16.9805881Z',
      id: 'a',
      isFavorite: false
    },
    {
      author: 'Douglas Main',
      title: 'Why insect populations are plummeting',
      description: 'Sad insects.',
      url:
        'https://www.nationalgeographic.com/animals/2019/02/why-insect-populations-are-plummeting-and-why-it-matters.html',
      urlToImage:
        'https://www.nationalgeographic.com/content/dam/animals/2019/02/insects-in-decline/01-insects-in-decline-nationalgeographic_1353541.jpg',
      publishedAt: '2019-02-19T02:07:15.6368398Z',
      id: 'b',
      isFavorite: false
    }
  ]
};

export const mockNewScienceArticlesObj = {
  articles: [
    {
      author: 'Chelsea Whyte',
      title: 'Wolves are spreading in Germany',
      description: 'Wolves everywhere!',
      url:
        'https://www.nationalgeographic.https://www.newscientist.com/article/2194696-wolves-are-spreading-in-germany-by-moving-into-military-land//animals/2019/02/why-insect-populations-are-plummeting-and-why-it-matters.html',
      urlToImage:
        'https://images.newscientist.com/wp-content/uploads/2019/02/22145340/rm7694.jpg',
      publishedAt: '2019-02-23T00:00:00Z',
      id: 'c',
      isFavorite: false
    },
    {
      author: 'Leah Crane',
      title: 'Four stars full of burnt nuclear ash',
      description: 'Scary stars!',
      url:
        'https://www.newscientist.com/article/2194683-four-stars-full-of-burnt-nuclear-ash-are-speeding-through-the-galaxy/',
      urlToImage:
        'https://images.newscientist.com/wp-content/uploads/2019/02/22150921/vennes1hr.jpg',
      publishedAt: '2019-02-22T00:00:00Z',
      id: 'd',
      isFavorite: false
    }
  ]
};

export const mockCryptoCoinsArticlesObj = {
  articles: [
    {
      author: 'Joseph Young',
      title: 'Bitcoins spike',
      description: 'So much money!',
      url:
        'https://www.ccn.com/bitcoin-price-4100-2-hours-analyst-crypto-bullish',
      urlToImage:
        'https://www.ccn.com/wp-content/uploads/2019/02/bitcoin-price-green-chart-bull-shutterstock.jpg',
      publishedAt: '2019-02-23T14:45:23Z',
      id: 'e',
      isFavorite: false
    },
    {
      author: 'Greg Thompson',
      title: 'Ethereum Slayers',
      description: 'They are everywhere.',
      url: 'https://www.ccn.com/ethereum-killers-crypto-blockchain',
      urlToImage:
        'https://www.ccn.com/wp-content/uploads/2019/02/ethereum-crypto-dark-cement-shutterstock.jpg',
      publishedAt: '2019-02-23T14:28:21Z',
      id: 'f',
      isFavorite: false
    }
  ]
};

export const mockNewSciArticlesStringified = {
  json: {
    articles: [
      {
        author: 'Chelsea Whyte',
        description: 'Wolves everywhere!',
        id: 'c',
        isFavorite: false,
        publishedAt: '2019-02-23T00:00:00Z',
        title: 'Wolves are spreading in Germany',
        url:
          'https://www.nationalgeographic.https://www.newscientist.com/article/2194696-wolves-are-spreading-in-germany-by-moving-into-military-land//animals/2019/02/why-insect-populations-are-plummeting-and-why-it-matters.html',
        urlToImage:
          'https://images.newscientist.com/wp-content/uploads/2019/02/22145340/rm7694.jpg'
      },
      {
        author: 'Leah Crane',
        description: 'Scary stars!',
        id: 'd',
        isFavorite: false,
        publishedAt: '2019-02-22T00:00:00Z',
        title: 'Four stars full of burnt nuclear ash',
        url:
          'https://www.newscientist.com/article/2194683-four-stars-full-of-burnt-nuclear-ash-are-speeding-through-the-galaxy/',
        urlToImage:
          'https://images.newscientist.com/wp-content/uploads/2019/02/22150921/vennes1hr.jpg'
      }
    ]
  },
  ok: true
};
