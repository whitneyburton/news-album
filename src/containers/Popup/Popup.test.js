import React from 'react';
import { shallow } from 'enzyme';
import { Popup } from './Popup';
import { mockNatGeoArticle } from '../../mockData';

const mockProps = { currentArticle: mockNatGeoArticle };
const mockMatch = { path: 'https://localhost.com/national-geographic/:id' };

describe('Popup', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Popup {...mockProps} match={mockMatch} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have the appropriate default state', () => {
    expect(wrapper.state('copied')).toEqual(false);
  });

  describe('CopyToClipboard component', () => {
    it('should set the state of copied to true', () => {
      wrapper.find('.CopyToClipboard--button').simulate('copy');
      expect(wrapper.state('copied')).toEqual(true);
    });
  });

  describe('toggleCopied', () => {
    it('should return COPY when the the copied state property is false', () => {
      const expected = 'COPY';
      const result = wrapper.instance().toggleCopied();
      expect(result).toEqual(expected);
    });

    it('should return COPIED! when the copied state property is true', () => {
      wrapper.setState({ copied: true });
      const expected = 'COPIED!';
      const result = wrapper.instance().toggleCopied();
      expect(result).toEqual(expected);
    });
  });
});