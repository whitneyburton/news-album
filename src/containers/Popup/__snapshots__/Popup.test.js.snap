// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup should match the snapshot 1`] = `
<div
  className="Popup"
>
  <Link
    replace={false}
    to="https://localhost.com/national-geographic"
  >
    <img
      alt="close icon"
      className="Popup--close"
      src="close.svg"
    />
  </Link>
  <img
    alt="star icon"
    className="Popup--star"
    src="favstar.svg"
  />
  <div
    className="Popup--info"
  >
    <div
      className="Popup--facts"
    >
      <h4
        className="Popup--title"
      >
        LIONS AND TIGERS AND BEARS
      </h4>
      <p
        className="Popup--author-pub"
      >
        , 
        Mon Feb 18 2019
      </p>
      <p
        className="Popup--description"
      />
    </div>
    <div
      className="Popup--buttons"
    >
      <a
        href="www.nationalgeographic.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        <button>
          READ
        </button>
      </a>
      <CopyToClipboard
        className="CopyToClipboard--button"
        onCopy={[Function]}
        text="www.nationalgeographic.com"
      >
        <button>
          COPY
        </button>
      </CopyToClipboard>
    </div>
  </div>
  <img
    alt="article"
    className="Popup--image"
    src="www.image.com"
  />
</div>
`;
